lib/rspec/core/formatters/documentation_formatter.rb

```diff
@@ -27,19 +34,46 @@ def example_group_finished(_notification)
 
         def example_passed(passed)
           output.puts passed_output(passed.example)
+
+          flush_messages
+          @example_started = false
         end
 
         def example_pending(pending)
           output.puts pending_output(pending.example,
                                      pending.example.execution_result.pending_message)
+
+          flush_messages
+          @example_started = false
         end
 
         def example_failed(failure)
           output.puts failure_output(failure.example)
+
+          flush_messages
+          @example_started = false
+        end
+
+        def message(notification)
+          if @example_started
+            @messages << notification.message
+          else
+            output.puts "#{current_indentation}#{notification.message}"
+          end
         end
 
       private
 
+        def flush_messages
+          if @messages.any?
```

Comment:
You don't need to check for this, you can just iterate and clear, its a no op on an empty array.